File: RESTAURANT/src/main/java/com/poly/restaurant/dtos/ReservationResponseDTO.java
package com.poly.restaurant.dtos;

import com.poly.restaurant.entities.enums.ReservationStatus;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class ReservationResponseDTO {

    private Long id;
    private LocalDateTime reservationTime;
    private int numberOfGuests;
    private String notes;
    private ReservationStatus status;
    private java.time.LocalDateTime createdAt;
    private java.time.LocalDateTime updatedAt;

    // Trả về thông tin cơ bản, tránh trả về cả object lồng nhau phức tạp
    private String customerName;
    private String tableName;
    private String branchName;
}

File: FE-RESTAURANT/src/interfaces/ReservationResponseDTO.ts
import { type ReservationStatus } from './enums/ReservationStatus';

export interface ReservationResponseDTO {
  id: number;
  reservationTime: string;
  numberOfGuests: number;
  notes: string;
  status: ReservationStatus;
  createdAt: string;
  updatedAt: string;
  customerName: string;
  tableName: string;
  branchName: string;
} 